From 9126f9fa3c88e30b34500de474b56dbca7ba1900 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Erkka=20K=C3=A4=C3=A4ri=C3=A4?= <erkka.kaaria@intel.com>
Date: Mon, 14 Dec 2015 14:40:29 +0200
Subject: [PATCH] icedtea7-native: Build hacks for ecj 3.8.2
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

ecj 3.8.2 is unable to build icedtea7-native without stubbing out unimplemented 
ssl functionality

Signed-off-by: Erkka Kääriä <erkka.kaaria@intel.com>
---
 .../share/classes/com/sun/net/httpserver/HttpsConfigurator.java    | 2 +-
 .../share/classes/sun/management/jmxremote/ConnectorBootstrap.java | 7 +------
 .../jdk/src/share/classes/sun/net/httpserver/SSLStreams.java       | 2 +-
 3 files changed, 3 insertions(+), 8 deletions(-)

diff --git a/openjdk/jdk/src/share/classes/com/sun/net/httpserver/HttpsConfigurator.java b/openjdk/jdk/src/share/classes/com/sun/net/httpserver/HttpsConfigurator.java
index 2543a15..645badb 100644
--- openjdk/jdk/src/share/classes/com/sun/net/httpserver/HttpsConfigurator.java
+++ openjdk/jdk/src/share/classes/com/sun/net/httpserver/HttpsConfigurator.java
@@ -110,7 +110,7 @@ public class HttpsConfigurator {
     * @since 1.6
     */
     public void configure (HttpsParameters params) {
-        params.setSSLParameters (getSSLContext().getDefaultSSLParameters());
+        throw new UnsupportedOperationException("Not implemented");
     }
 //END_TIGER_EXCLUDE
 }
diff --git a/openjdk/jdk/src/share/classes/sun/management/jmxremote/ConnectorBootstrap.java b/openjdk/jdk/src/share/classes/sun/management/jmxremote/ConnectorBootstrap.java
index 3e48089..5dfac3a 100644
--- openjdk/jdk/src/share/classes/sun/management/jmxremote/ConnectorBootstrap.java
+++ openjdk/jdk/src/share/classes/sun/management/jmxremote/ConnectorBootstrap.java
@@ -657,12 +657,7 @@ public final class ConnectorBootstrap {
 
                 SSLContext ctx = SSLContext.getInstance("SSL");
                 ctx.init(kmf.getKeyManagers(), tmf.getTrustManagers(), null);
-
-                return new SslRMIServerSocketFactory(
-                        ctx,
-                        enabledCipherSuites,
-                        enabledProtocols,
-                        sslNeedClientAuth);
+                throw new UnsupportedOperationException("Not implemented");
             } catch (Exception e) {
                 throw new AgentConfigurationError(AGENT_EXCEPTION, e, e.toString());
             }
diff --git a/openjdk/jdk/src/share/classes/sun/net/httpserver/SSLStreams.java b/openjdk/jdk/src/share/classes/sun/net/httpserver/SSLStreams.java
index c6b87cd..2a08a87 100644
--- openjdk/jdk/src/share/classes/sun/net/httpserver/SSLStreams.java
+++ openjdk/jdk/src/share/classes/sun/net/httpserver/SSLStreams.java
@@ -78,7 +78,7 @@ class SSLStreams {
             cfg.configure (params);
             SSLParameters sslParams = params.getSSLParameters();
             if (sslParams != null) {
-                engine.setSSLParameters (sslParams);
+                throw new UnsupportedOperationException("Not implemented");
             } else
 //END_TIGER_EXCLUDE
             {
-- 
2.1.4

